pkg/gardenlet/controller/networkpolicy/reconciler.go

```diff
@@ -34,35 +33,23 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 )
 
-// namespaceReconciler implements the reconcile.Reconcile interface for namespace reconciliation.
-type namespaceReconciler struct {
-	seedClient             client.Client
-	seedName               string
-	shootNamespaceSelector labels.Selector
-	resolver               hostnameresolver.HostResolver
-}
+// Reconciler implements the reconcile.Reconcile interface for namespace reconciliation.
+type Reconciler struct {
+	SeedClient           client.Client
+	Config               config.SeedAPIServerNetworkPolicyControllerConfiguration
+	Resolver             hostnameresolver.HostResolver
+	GardenNamespace      *corev1.Namespace
+	IstioSystemNamespace *corev1.Namespace
 
-// newNamespaceReconciler returns the new namespace reconciler.
-func newNamespaceReconciler(
-	seedClient client.Client,
-	seedName string,
-	shootNamespaceSelector labels.Selector,
-	resolver hostnameresolver.HostResolver,
-) reconcile.Reconciler {
-	return &namespaceReconciler{
-		seedClient:             seedClient,
-		seedName:               seedName,
-		shootNamespaceSelector: shootNamespaceSelector,
-		resolver:               resolver,
-	}
+	shootNamespaceSelector labels.Selector
```

Comment:
As mentioned earlier, this does not need to be a string attribute, does it? We can declare the selector above and just use it?

Reply:
I removed `shootNamespaceSelector` from `Reconciler`, but I'm not sure if this is what you suggested 😅